aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/BaseHandlerStd.java

```diff
@@ -30,4 +44,84 @@
+              "ExternalConnections for repositories are currently limited to 1.");
+      }
+
+      // Loop is currently not necessary because only 1 external connection is allowed, leaving this in for
+      // when multiple are supported
+      externalConnectionsToAdd.forEach(ec -> {
+          try {
+              AssociateExternalConnectionRequest associateExternalConnectionRequest
+                  = Translator.translateAssociateExternalConnectionsRequest(resourceModel, ec);
+
+              proxyClient.injectCredentialsAndInvokeV2(associateExternalConnectionRequest, proxyClient.client()::associateExternalConnection);
+          } catch (final AwsServiceException e) {
+              String repositoryName = progress.getResourceModel().getRepositoryName();
+              Translator.throwCfnException(e, Constants.ASSOCIATE_EXTERNAL_CONNECTION, repositoryName);
+          }
+          logger.log(String.format("Successfully associated external connection: %s", ec));
+      });
+
+      return ProgressEvent.<ResourceModel, CallbackContext>builder()
+          .resourceModel(resourceModel)
+          .status(OperationStatus.IN_PROGRESS)
+          .build();
+  }
+
+  protected ProgressEvent<ResourceModel, CallbackContext> putRepositoryPermissionsPolicy(
+      final AmazonWebServicesClientProxy proxy,
+      final ProgressEvent<ResourceModel, CallbackContext> progress,
+      final CallbackContext callbackContext,
+      final ProxyClient<CodeartifactClient> proxyClient,
+      final Logger logger
+  ) {
+      final ResourceModel desiredModel = progress.getResourceModel();
+      if (desiredModel.getPolicyDocument() == null) {
+          return ProgressEvent.progress(desiredModel, callbackContext);
+      }
+      return proxy.initiate("AWS-CodeArtifact-Repository::Update::PutRepositoryPermissionsPolicy", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+          .translateToServiceRequest(Translator::translatePutPermissionsPolicyRequest)
+          .makeServiceCall((awsRequest, client) -> {
+              PutRepositoryPermissionsPolicyResponse awsResponse = null;
+              try {
+                  awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::putRepositoryPermissionsPolicy);
+                  logger.log("Repository permission policy successfully added.");
+              } catch (final AwsServiceException e) {
+                  String domainName = desiredModel.getDomainName();
+                  Translator.throwCfnException(e, Constants.PUT_REPOSITORY_POLICY, domainName);
+              }
+              return awsResponse;
+          })
+          .stabilize((awsRequest, awsResponse, client, model, context) -> policyIsStabilized(model, client))
+          .progress();
+  }
+
+  private boolean policyIsStabilized(final ResourceModel model, final ProxyClient<CodeartifactClient> proxyClient) {
```

Comment:
Do we need to `stabilize` for policy updates? I'm not familiar enough with the CFN libraries to know what the purpose of `stabilize` is.

Is this because you think that updating the permission policy is eventually consistent? When performing authorization it is eventually consistent (after updating a policy it may take a few minutes before that policy is used when making authorization decisions). But from the perspective of put and get permission policy operations it is strongly consistent. 

So it is unclear to me why we need the call to GetRepositoryPermissionsPolicy. And if we think we need the call, then why don't we check the return value to make sure it matches the desired policy?

Reply:
If the operation is strong consistent, then I agree that we don't need to `stabilize` here. Will update